src/App.jsx

```diff
@@ -13,46 +15,84 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const SORTED_ALPHABETICALLY = 'Sort alphabetically';
+const SORTED_BY_LENGTH = 'Sort by length';
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [value, setValue] = useState('');
+  const [reversed, setReversed] = useState(false);
+  let sortedGoods = [...goodsFromServer];
+
+  switch (value) {
+    case SORTED_ALPHABETICALLY: sortedGoods
+      .sort((a, b) => a.localeCompare(b));
+      break;
+    case SORTED_BY_LENGTH: sortedGoods
+      .sort((a, b) => a.length - b.length);
+      break;
+    default: sortedGoods = [...goodsFromServer];
+      break;
+  }
+
+  if (reversed) {
+    sortedGoods.reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn({ 'is-light': value !== SORTED_ALPHABETICALLY },
+            'button', 'is-info')}
+          onClick={value === SORTED_ALPHABETICALLY
+            ? () => setValue('')
+            : () => setValue(SORTED_ALPHABETICALLY)}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={cn({ 'is-light': value !== SORTED_BY_LENGTH },
+            'button', 'is-success')}
+          onClick={value === SORTED_BY_LENGTH
+            ? () => setValue('')
+            : () => setValue(SORTED_BY_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn({ 'is-light': !reversed },
+            'button', 'is-warning')}
+          onClick={reversed
+            ? () => setReversed(false)
```

Comment:
you can write here setReversed(!reversed)

Reply:
Better to use `functional set state` e.g `onClick={() => setReversed(current => !current)}`